src/model_theory/definability.lean

```diff
@@ -45,6 +45,47 @@ structure definable (s : set (α → M)) : Prop :=
 
 variables {L} {A} {B : set M} {s : set (α → M)}
 
+lemma definable.map_expansion {L' : language} [L'.Structure M] (h : L.definable A s)
+  (φ : L →ᴸ L') [φ.is_expansion_on M] :
+  L'.definable A s :=
+begin
+  obtain ⟨ψ, rfl⟩ := h,
+  refine ⟨⟨(φ.add_constants A).on_formula ψ, _⟩⟩,
+  ext x,
+  simp only [mem_set_of_eq, Lhom.realize_on_formula],
+end
+
+lemma empty_definable_iff :
+  L.definable ∅ s ↔ ∃ (φ : L.formula α), s = set_of φ.realize :=
```

Comment:
With some more basic properties, this proof can be very short.
I think it's nice to add
(1)
```lean
lemma definable_iff : L.definable A s ↔ ∃ (φ : L[[A]].formula α), s = set_of φ.realize :=
⟨λ h, h.1, λ h, ⟨h⟩⟩
```
[you could also make `definable` a `def`]
(2) a version of `Lhom_trim_empty_constants` that is an `language.equiv`
(3) `language.equiv` induces equivalences on `term` and `[bounded_]formula`
Then this follows immediately from `equiv.exists_congr_left`.

(it's not that the current proof is particularly hard, but it shows that there is missing API in the more basic files)

Reply:
Oh yes, I forgot that `Lhom` and `language.equiv` are on different "levels".
Then you can wait with these for a future PR, but I think it will still be useful to simplify more future arguments.

Reply:
Sounds good.

Reply:
#12837